core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val stateLock = new ReentrantLock
 
   val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
+  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
```

Comment:
@klueska  I think you need to autodiscover anything, the concept is similar to cpus max in the scheduler.
@tnachen  I think there should be some logic checking current total against the max gpus configured like in the case of cpusMax. I dont see any. I expect offers to be splitted. In that case we need to check the sum of the assigned gpus against the max right?